src/board_controller/pieeg/pieeg_board.cpp

```diff
@@ -0,0 +1,364 @@
+        server_socket = nullptr;
+    }
+
+    return (int)BrainFlowExitCodes::STATUS_OK;
+}
+
+int PIEEGBoard::config_board(std::string config, std::string &response)
+{
+    if (!initialized)
+    {
+        return (int)BrainFlowExitCodes::BOARD_NOT_READY_ERROR;
+    }
+    return (int)BrainFlowExitCodes::UNSUPPORTED_BOARD_ERROR;
+}
+
+void PIEEGBoard::read_thread()
+{
+    uint8_t buf[27] = {0};
+    uint8_t zero27[27] = {0};
+    int num_rows = board_descr["default"]["num_rows"];
+    double *package = new double[num_rows];
+    for (int i = 0; i < num_rows; i++)
+    {
+        package[i] = 0.0;
+    }
+
+    std::vector<int> eeg_channels = board_descr["default"]["eeg_channels"];
+
+    double eeg_scale = 4.5 / float((pow(2, 23) - 1)) / 8 * 1000000.;
+    double timestamp = 0;
+    int counter = 0;
+    int timeout_ms = 1000;
+    uint32_t data_test = 0x7FFFFF;
+    uint32_t data_check = 0xFFFFFF;
+
+    while (keep_alive)
+    {
+        int gpio_res = gpio_poll(gpio_in, timeout_ms);
+        timestamp = get_timestamp();
+        if (gpio_res == 0)
+        {
+            safe_logger(spdlog::level::trace, "no gpio event in {} ms", timeout_ms);
+        }
+        else if (gpio_res < 0)
+        {
+            safe_logger(spdlog::level::warn, "error in gpio_poll: {}", gpio_res);
+        }
+        else
+        {
+            gpio_edge_t edge = GPIO_EDGE_NONE;
+            gpio_res = gpio_read_event(gpio_in, &edge, NULL);
+            if (gpio_res != 0)
+            {
+                safe_logger(spdlog::level::warn, "failed to get gpio event: {}", gpio_res);
+                continue;
+            }
+            if (edge != GPIO_EDGE_FALLING)
+            {
+                safe_logger(spdlog::level::warn, "unexpected gpio event");
+                continue;
+            }
+            int spi_res = spi_transfer(spi, zero27, buf, 27);
+            if (spi_res != 0)
+            {
+                safe_logger(spdlog::level::warn, "failed to read from spi: {}", spi_res);
+                continue;
+            }
+            for (size_t i = 0; i < eeg_channels.size(); i++)
+            {
+                int offset = 3 * i + 3;
+                uint32_t voltage = (buf[offset] << 8) | buf[offset + 1];
+                voltage = (voltage << 8) | buf[offset + 2];
+                uint32_t voltage_test = voltage | data_test;
+                if (voltage_test == data_check)
+                {
+                    voltage = 16777214 - voltage;
+                }
+
+                package[eeg_channels[i]] = 0.27 * voltage;
+            }
+            package[board_descr["default"]["timestamp_channel"].get<int>()] = timestamp;
+            package[board_descr["default"]["package_num_channel"].get<int>()] = counter++;
+            push_package(package);
+
+            // Send data over TCP
+            if (server_socket && server_socket->client_connected)
+            {
+                server_socket->send(package, num_rows * sizeof(double));
```

Comment:
it should not send anything from here, idea of this code is to read data straight on raspberry pi and if its needed to be sent to PC or somewhere use high level API like streaming boards, or user's own implementations

Reply:
Removed the custom data sending code as streaming should be done via high-level APIs.